stablehlo/reference/Tensor.h

```diff
@@ -99,6 +109,11 @@ class Tensor {
 
  private:
   llvm::IntrusiveRefCntPtr<detail::Buffer> impl_;
+  std::vector<int64_t> strides_;
```

Comment:
We are computing the strides from the type using a free function `computeStride` defined in Tensor.cpp

Reply:
Why do we need to store strides if we don't provide an API to change them? Can we compute them on the fly? This way we'll be sure that they are always what we expect, and this will allow constructors to have default implementations again. 

Also, as a bonus, since strides are computed only for the purposes of flattening indices, you won't even need to have a dedicated function for that. You'll be able to inline this computation in the flattening logic.